### ISSUE TITLE
Documentation on the best way to nest providers inside other providers

### ISSUE THREAD

**Issue by not-holar** [rrousselGit/river_pod#66]
**Describe what scenario you think is uncovered by the existing examples/articles**
Sometimes you want to nest providers inside of other providers for e.g. a listenable list where you want to only listen to item reordering, adding and removing, while all the internal items are individually listenable, so that you can avoid rebuilding everything when only a single item changes. What would be the most idiomatic way of achieving this in Riverpod?

**Comment by rrousselGit** [comment]
I don't really understand what this is about.
Could you explain more?

**Comment by not-holar** [comment]
Let's say we took something like the Todo example, but made each [TodoItem](https://github.com/rrousselGit/river_pod/blob/00d4e976ff29a404f030c50857c7ae9d0eaa6026/examples/todos/lib/main.dart#L212) widget much more complex, so rebuilding them all at once when we change a field of only one item would be inefficient.

Right now all changes to a Todo are done through [TodoList](https://github.com/rrousselGit/river_pod/blob/00d4e976ff29a404f030c50857c7ae9d0eaa6026/examples/todos/lib/todo.dart#L26) and you can only listen to the whole list at once.

You could wrap each list item in something like ValueListenable and then each item widget in ValueListenableBuilder for more granular rebuilds, but that would bring in a flutter dependency.

Basically what i'm asking about is a more modular state, where items of a list could change and notify individually without rebuilding the whole list

**Comment by hpn33** [comment]
on provider that was easy for me
i make my item with `ChangeNotifier`
`class Item with ChangeNotfier {}`
and when building widget list i put them on `ChangeNotifierProvider.value`
```
ChangeNotifierProvider.value(
value: item,
builder: ....)
```
and its worked
when item is change just that widget on list going change

but in riverpod is complex

**Comment by rrousselGit** [comment]
One solution is to pass the index/id to the item, then use a `Computed.family` (or `Provider.family` on 0.6.0-dev):


```dart
ListView.builder(
  itemBuilder: (context, index) => TodoItem(index),
);

...

final todoItemProvider = Computed.family<Todo, int>((watch, index) {
  return watch(todoListProvider.state)[index];
});

...

final int index;
Widget build(context) {
  final item useProvider(todoItemProvider(index));
}
```

**Comment by arnoutvandervorst** [comment]
static final provider = StreamProvider.family
      .autoDispose<List<GroupEarning>, String>((_, groupId) {
    return GroupEarnings().streamList(groupId);
  });

This works on 0.6.0, very nice!

**Comment by rrousselGit** [comment]
The todo/marvel app showcase how you can avoid unnecessary item rebuild, so I will close this for now
